### PR TITLE
Use JSON for trait encoding.

### PR DESCRIPTION
**Pull request by russjones**
**Description**

Use JSON for trait encoding. Fallback to protobuf.

### CODE REVIEW

File: lib/wrappers/wrappers.go [hunk 1]
```diff
@@ -41,6 +41,26 @@ func (l Traits) protoType() *LabelValues {
...
 
+// MarshalTraits will marshal Traits as JSON. Used to embed traits into
+// certificates.
+func MarshalTraits(v interface{}) ([]byte, error) {
+	return json.Marshal(v)
+}
+
+// UnmarshalTraits will unmarshal JSON traits. Used to embed traits into
+// certificates.
+func UnmarshalTraits(data []byte, v interface{}) error {
```
Comment (klizhentas) [inline lib/wrappers/wrappers.go]: any reason why `v` is not of type `Traits` then?

File: lib/wrappers/wrappers.go [hunk 2]
```diff
@@ -41,6 +41,26 @@ func (l Traits) protoType() *LabelValues {
 	return v
 }
 
+// MarshalTraits will marshal Traits as JSON. Used to embed traits into
+// certificates.
+func MarshalTraits(v interface{}) ([]byte, error) {
```
Comment (klizhentas) [inline lib/wrappers/wrappers.go]: could it be `v Traits` type then?

**Review by klizhentas — APPROVED** [review]

**Review by fspmarshall — APPROVED** [review]
LGTM.  Might be nice to have a `DELETE IN 4.X` directive for the protobuf compat (though I'm not sure when that would be in this case :confused:).

**Comment by russjones** [comment]
@gravitational-jenkins retest this please

**Comment by russjones** [comment]
@gravitational-jenkins retest this please

**Pull request merged**